File: go/array/118_generate.go
package array

// 杨辉三角
func generate(numRows int) [][]int {
	result := make([][]int, numRows)

	for i := 0; i < numRows; i++ {
		result[i] = make([]int, i+1)
		// 每行第一个数字
		result[i][0] = 1
		// 每行最后一个数字
		result[i][i] = 1

		for j := 1; j < i; j++ {
			result[i][j] = result[i-1][j-1] + result[i-1][j]

		}
	}

	return result
}

File: rust/array/118_generate.rs



// 杨辉三角
pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {

    let mut result: Vec<Vec<i32>> = vec![];

    for i in 0..num_rows as usize {

        // 初始把一行数据全部赋值为1
        result.push(vec![1;i+1]);

        for j in 1..i {
            // 根据杨辉三角赋值
            result[i][j] = result[i-1][j-1] + result[i-1][j]
        }
    }

    result

}